Fix part of #2294: Migrating to TextInputLayout for AddProfileActivity

FareesHussain opened the pull request:
<!--
  - Thanks for submitting code to Oppia! Please fill out the following as part of
  - your pull request so we can review your code more easily.
  -->

## Explanation
<!--
  - Explain what your PR does. If this PR fixes an existing bug, please include
  - "Fixes #bugnum:" in the explanation so that GitHub can auto-close the issue
  - when this PR is merged.
  -->
Fix part of #2294 
Fix part of #962 
Migrating for `AddProfileActivity`

## Checklist
<!-- Please tick the relevant boxes by putting an "x" in them. -->
- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

## ScreenShots
Username Portrait | Username Landscape
--- | ---
![image](https://user-images.githubusercontent.com/54740946/103310014-9a2bff80-4a3c-11eb-9441-5eb01f71b5c0.png) | ![image](https://user-images.githubusercontent.com/54740946/103310032-a6b05800-4a3c-11eb-92ed-8f87ca0a5cd6.png)
![image](https://user-images.githubusercontent.com/54740946/103310101-ce072500-4a3c-11eb-9371-0c9e5ceed950.png) | ![image](https://user-images.githubusercontent.com/54740946/103310112-d7908d00-4a3c-11eb-9da1-24798ee0cd3b.png)
![image](https://user-images.githubusercontent.com/54740946/103310197-10c8fd00-4a3d-11eb-8591-ae825f690a86.png) | ![image](https://user-images.githubusercontent.com/54740946/103310221-19213800-4a3d-11eb-95a4-83dec8c6c841.png)

Pin Portrait | Pin Landscape
--- | ---
![image](https://user-images.githubusercontent.com/54740946/103310319-4f5eb780-4a3d-11eb-8629-63ca824b6160.png) | ![image](https://user-images.githubusercontent.com/54740946/103310345-5c7ba680-4a3d-11eb-8b34-2f4292413137.png)
![image](https://user-images.githubusercontent.com/54740946/103310366-6bfaef80-4a3d-11eb-929a-786c34664696.png) | ![image](https://user-images.githubusercontent.com/54740946/103310377-74532a80-4a3d-11eb-92b6-4107f366314f.png)
![image](https://user-images.githubusercontent.com/54740946/103310415-951b8000-4a3d-11eb-9c40-8e9ef010bcbc.png) | ![image](https://user-images.githubusercontent.com/54740946/103310435-9f3d7e80-4a3d-11eb-8f37-f802209d153f.png)

## Video

![addProfileActivity](https://user-images.githubusercontent.com/54740946/103310932-ea0bc600-4a3e-11eb-81da-27e199b34c0d.gif)

## Tests
**Espresso**
![image](https://user-images.githubusercontent.com/54740946/103329193-b2723d80-4a81-11eb-9122-93f16cab8975.png)

**Robolectric**
![image](https://user-images.githubusercontent.com/54740946/103329039-0d576500-4a81-11eb-9687-da2fd287e69e.png)

FareesHussain committed:
issue 2098

FareesHussain committed:
Merge branch 'develop' of https://github.com/oppia/oppia-android into develop

FareesHussain committed:
Migrating to TextInputLayout for AddProfileActivity

FareesHussain committed:
Reformatting code and removing comments

anandwana001 commented:
Same comment from other issue, add test screenshots.

anandwana001 commented on app/src/main/res/layout-land/add_profile_activity.xml:
```diff
@@ -119,36 +127,53 @@
               android:layout_height="wrap_content"
               android:layout_marginStart="28dp"
-              android:layout_marginTop="8dp"
+              android:layout_marginTop="24dp"
```
is the new margin according to the mocks?

FareesHussain replied:
No, I'll revert that

FareesHussain replied:
Done

anandwana001 commented on app/src/main/res/layout-sw600dp-land/add_profile_activity.xml:
```diff
@@ -73,18 +72,27 @@
+              android:layout_height="wrap_content"
+              android:hint="@string/add_profile_input_name"
+              android:maxLength="36"
+              android:singleLine="true"
```
reformat and re-arrange.

FareesHussain replied:
Done

anandwana001 reviewed (changes requested):
Thanks @FareesHussain Doing a quick review
We need test cases covering pressing of IME action button.

FareesHussain commented:
@anandwana001 I have a doubt regarding IME_ACTION in this activity
for username i've added IME_ACTION_DONE and pressing it closes the keyboard
for pin i've added IME_ACTION_NEXT and pressing it the focus goes to confirm pin
for confirm pin i've added IME_ACTION_DONE and pressing the creates a new profile


It looks good or do i have to make changes for the IME ACTIONS

FareesHussain committed:
Reformatting xml

anandwana001 commented:
> @anandwana001 I have a doubt regarding IME_ACTION in this activity
> for username i've added IME_ACTION_DONE and pressing it closes the keyboard
> for pin i've added IME_ACTION_NEXT and pressing it the focus goes to confirm pin
> for confirm pin i've added IME_ACTION_DONE and pressing the creates a new profile
> 
> It looks good or do i have to make changes for the IME ACTIONS

Defer to @rt4914 
Closing keyboard from name sounds ok to me but if the check is marked then we can shift the focus to pin from the name, else the keyboard will get close.

FareesHussain committed:
RTL and textInputLayout hint updated

rt4914 commented on app/src/main/java/org/oppia/android/app/profile/AddProfileActivityPresenter.kt:
```diff
@@ -112,43 +113,48 @@ class AddProfileActivityPresenter @Inject constructor(
+
+    binding.addProfileActivityPinEditText.onTextChanged { pin ->
+      pin?.let {
+        profileViewModel.inputPin.set(it.toString())
```
You don't need `toString()`

FareesHussain replied:
Done

rt4914 commented on app/src/main/java/org/oppia/android/app/profile/AddProfileActivityPresenter.kt:
```diff
@@ -112,43 +113,48 @@ class AddProfileActivityPresenter @Inject constructor(
+
+    binding.addProfileActivityConfirmPinEditText.onTextChanged { confirmPin ->
+      confirmPin?.let {
+        profileViewModel.inputConfirmPin.set(it.toString())
```
You don't need `toString()`

FareesHussain replied:
Done

rt4914 reviewed (approved):
LGTM, thanks.

FareesHussain committed:
Resolving requested changes

FareesHussain commented:
@rt4914 please re-run the workflow
![image](https://user-images.githubusercontent.com/54740946/103408770-f6e90080-4b89-11eb-9a85-779f2d8cfd62.png)

FareesHussain committed:
Merge branch 'develop' of https://github.com/oppia/oppia-android into textInputLayout-in-AddProfile

FareesHussain commented:
@rt4914 PTAL

FareesHussain committed:
test commit

FareesHussain committed:
Merge branch 'develop' of https://github.com/FareesHussain/oppia-android into develop

FareesHussain committed:
Update gradle.xml

FareesHussain committed:
Update misc.xml

FareesHussain committed:
Fix gradle.xml

FareesHussain committed:
Merge branch 'develop' of https://github.com/FareesHussain/oppia-android into develop

FareesHussain committed:
Merge branch 'develop' of https://github.com/FareesHussain/oppia-android into develop

FareesHussain committed:
Merge branch 'develop' of https://github.com/FareesHussain/oppia-android into textInputLayout-in-AddProfile

FareesHussain committed:
Databinding for text

rt4914 reviewed (approved):
LGTM, thanks.

anandwana001 reviewed (approved):
LGTM, Thanks @FareesHussain
